autogen/main/versions.tf.tmpl

```diff
@@ -24,11 +24,11 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = ">= 5.40.0, < 7"
+      version = ">= 5.44.2, !=6.0.0, !=6.0.1, !=6.1.0, !=6.2.0, !=6.3.0, !=6.4.0, !=6.5.0, !=6.6.0, < 7"
```

Comment:
I didn't see any patch releases between 6.1 and 6.6, but let me know if I missed something.
An alternate approach might be to allow some of the other 6.x ones that are not completely broken, but requiring either 5.44.2 or 6.7.0 is probably the safest.

Reply:
Thanks @wyardley - I agree this is the least error prone for now, and in the subsequent breaking release we can drop support for v5 (for https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/2126), which will again simplify the version constraint.